autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/UpdateFailedHandler.java

```diff
@@ -51,15 +59,35 @@ public void onApplicationEvent(UpdateFailedEvent event) {
         if (failed < RETRY_THRESHOLD) {
             updateFailures.put(autoscaleClusterId, failed);
             LOGGER.debug("Increased failed count '{}' for cluster '{}'", failed, cluster.getStackCrn());
+            // TODO: Determine what needs to be done if ForbiddenException occurs with endUser.
```

Comment:
Nothing for now. Treat it as a regular failure, which is what we were doing. There's not a lot we can do to recover, other than being more aggressive about re-initializing the machine user, or having a better console message.